gateway/handlers/reader.go

```diff
@@ -52,6 +64,7 @@ func MakeFunctionReader(metricsOptions metrics.MetricOptions, c *client.Client)
 					Image:           service.Spec.TaskTemplate.ContainerSpec.Image,
 					InvocationCount: 0,
 					Replicas:        *service.Spec.Mode.Replicated.Replicas,
+					AvailableCount:  running[service.ID],
```

Comment:
I'm not sure this is safe, shouldn't we check if service.ID is a member of the collection first?

Reply:
It will be `nil` if `service.ID` is not in the `running`. `AvailableCount` would then be set to `nil`. The only time `AvailableCount` is used is to print the value, so `nil` should be safe. However, it would improve readability to have an `if` statement above that checks if this value is nil, and if so set `AvailableCount` to 0.